execution_engine/src/core/engine_state/mod.rs

```diff
@@ -1961,7 +1961,8 @@ where
         let gas_limit = Gas::new(U512::from(std::u64::MAX));
         let deploy_hash = {
             // seeds address generator w/ era_end_timestamp_millis
-            let bytes = step_request.era_end_timestamp_millis.into_bytes()?;
+            let mut bytes = step_request.era_end_timestamp_millis.into_bytes()?;
+            bytes.append(&mut step_request.next_era_id.into_bytes()?);
```

Comment:
I think this is unnecessary, I don't see how it's adding more entropy to the seed.

Reply:
It's not possible to receive two `StepRequests` for the same era - we only have it when it's a switch block and we're guaranteed to have only one of these.

Reply:
Well thats how its currently implemented, you could imagine two nodes making `StepRequests` to the same EE instance given that the EE has no notion of the blockchain itself. This IMO makes the seed generation more robust.

Reply:
Those two blockchains would have to have exactly same performance, down to a milliseconds of block productions since the genesis.

Reply:
Sure, what I am trying to say here is that how we pass in the `StepRequest` to the EE itself is part of how the node interacts with it, and that may/may not change. This adds additional variability in the seed.

@Fraser999 or @mpapierski do you guys have any opinions here?

Reply:
It's still not guaranteed that adding a `next_era_id` increases security as there's no way to guarantee that it will be different from "the other `StepRequest`'s".

Reply:
If you want to provide more randomness, use `step_request.pre_state_hash`. If that's _still_ equivalent then it should be fine b/c what we have is basically a function `f(x)` where `f` is the logic and `x` is the prestate hash. If both `f` and `x` are equivalent then the result also will be. Meaning it will be a no-op from the external pov.

Reply:
Yes, not by itself, however, the combination of both `next_era_id` and the timestamp for two different `StepRequests` is very unlikely to be the same.

Reply:
I'm fine leaving the code as per authored here and continuing this discussion offline.